cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/partitioned/ConsumablePartition.java

```diff
@@ -54,7 +54,7 @@
   void take();
 
   /**
-   * Marks the ProcessState as AVAILABLE, resets the timestamp to 0, and increments the number of failures by 1
+   * Marks the ProcessState as AVAILABLE and resets the timestamp to 0
```

Comment:
Wouldn't it be fine, since:
1) PartitionConsumer APIs are @Beta
2) While they technically could, I don't think users would/have override ConsumablePartition and WorkingSet classes.

The only reason is that then we have this new (almost redundant) method, `retryWithoutFailureCount`